app/src/Navigators/Main.tsx

```diff
@@ -1,62 +1,144 @@
+import {
+  ProfileContainer,
+  VerticalStackContainer,
+} from '@/NativeBase/Containers'
+import { EventTicket, Home, Projects, User } from '@/NativeBase/Assets/Icons'
+import { View, useColorMode, ColorMode } from 'native-base'
+import StaTheme from '@/NativeBase/Theme/StaTheme'
+
+type navBarProps = {
+  focused: boolean
+  colorMode: ColorMode
+}
 
 const Tab = createBottomTabNavigator()
 
-// @refresh reset
+const TabBarIconWrapper = ({ focused, colorMode }: navBarProps) => (
+  <View
+    borderTopColor={
+      focused
+        ? colorMode === 'light'
+          ? StaTheme.colors.primary['100']
+          : StaTheme.colors.primary['40']
+        : 'transparent'
+    }
+    borderTopWidth={2}
+    style={{
+      width: 97,
+    }}
+  />
+)
+
+const iconColor = ({ focused, colorMode }: navBarProps) => {
+  return focused
+    ? colorMode === 'light'
+      ? StaTheme.colors.primary['100']
+      : StaTheme.colors.primary['40']
+    : colorMode === 'light'
+    ? StaTheme.colors.text['100']
+    : StaTheme.colors.textDarkMode['100']
+}
+
+const HomeTabBarIcon = ({ focused, colorMode }: navBarProps) => (
+  <>
+    <TabBarIconWrapper focused={focused} colorMode={colorMode} />
+    <View
+      style={{
+        padding: 4,
+      }}
+    >
+      <Home color={iconColor({ focused, colorMode })} />
+    </View>
+  </>
+)
+const ProjectsTabBarIcon = ({ focused, colorMode }: navBarProps) => (
+  <>
+    <TabBarIconWrapper focused={focused} colorMode={colorMode} />
+    <View
+      style={{
+        padding: 9,
+      }}
+    >
+      <Projects color={iconColor({ focused, colorMode })} />
+    </View>
+  </>
+)
+const EventsTabBarIcon = ({ focused, colorMode }: navBarProps) => (
+  <>
+    <TabBarIconWrapper focused={focused} colorMode={colorMode} />
+    <View
+      style={{
+        padding: 7,
+      }}
+    >
+      <EventTicket color={iconColor({ focused, colorMode })} />
+    </View>
+  </>
+)
+const ProfileTabBarIcon = ({ focused, colorMode }: navBarProps) => (
+  <>
+    <TabBarIconWrapper focused={focused} colorMode={colorMode} />
+    <View
+      style={{
+        padding: 4,
+      }}
+    >
+      <User color={iconColor({ focused, colorMode })} />
+    </View>
+  </>
+)
+
+/**
+ * @returns {import('@react-navigation/bottom-tabs').BottomTabNavigator} A bottom tab navigator component from the '@react-navigation/bottom-tabs' package
+ */
 const MainNavigator = () => {
-  const { Colors } = useTheme()
+  const { colorMode } = useColorMode()
+
   return (
     <Tab.Navigator>
       <Tab.Screen
-        name="Projects"
-        component={ListContainer}
-        initialParams={{ type: ListType.Projects }}
+        name="Example Container"
+        component={VerticalStackContainer}
         options={{
-          headerShown: false,
-          tabBarIcon: () => (
-            <Entypo color={Colors.text} name="home" size={24} />
+          tabBarIcon: ({ focused }) => (
```

Comment:
We're getting a React warning message here "Do not define components during render."  Would be good to address this (and for the other `tabBarIcon`s).

I think what you've got here is nearly right, but instead of calling a component (like `HomeTabBarIcon`), call a function which renders a component -- a bit like [this one.](https://github.com/Scottish-Tech-Army/Volunteer-app/commit/722b209d0e5d3bf5719770b492cfe9df3be5c683#diff-08c2d3e3529a4185692086d551e348cf1ea0f234b50783e269e3372dc364547dR33)

Also, as the current `HomeTabBarIcon`, `ProjectsTabBarIcon`, etc mostly share the same code, would be good if a single render function could replace all of them and you could just pass in as parameters like focused, colorMode and icon (or a variable that can be used to determine which icon to use).

Give me a shout if that doesn't make sense or you need any help.

Reply:
Thanks @joefhall, all valid points, now resolved :)

Reply:
Oops, think it's you that's supposed to click resolved? ;)